api-http/proposals/snapshot-api.md

```diff
@@ -0,0 +1,566 @@
+# Snapshot Scheduling API Design specs
+
+:bulb: Following document embodies API proposal in relation to the following github issue https://github.com/AntelopeIO/leap/issues/396.
+
+## :beginner: Design Info
+
+### Preface
+It was requested that snapshot functionality and, in particular, snapshot API needs to be extended to support additional functionality. In particular, ability to schedule snapshot(s), edit schedule and query snapshot status.
+
+Following goals for an API were defined in mentioned above github issue:
+- Scenario 1: Create a snapshot for a specific block num
+- Scenario 2: Create a snapshot every X number of blocks
+
+Following considerations/design decisions were also expressed:
+- Ability to query existing requests and cancel them
+- Do pending requests persist across nodeos restarts?
+
+---
+
+
+:small_blue_diamond:Status:
+- [X] In progress
+- [ ] In review. Reviewer:
+- [ ] Finished
+---
+## :wrench: Proposal API additions
+
+
+| **API URL** 	| **Purpose** 	| **Parameters** 	|
+|----------	|----------	|----------	|
+| /producer/schedule_snapshot         	| Adds a task to perform snapshot         	| start_block, end_block, blocks_count, snapshot_description     	|
+| /producer/get_snapshot_requests         	| Returns scheduled snapshot requests         	| start_snapshot_request_id, limit (number of snapshots to return)         	|
+| /producer/get_snapshot_request_status         	| Queries status of a snapshot request        	| Snapshot ID         	|
+| /producer/unschedule_snapshot         	| Removes previously scheduled snapshot task         	| Snapshot ID         	|
+
+### Scheduling Snapshot
+- request takes 4 optional parameters - start_block, end_block, blocks_count, snapshot_description:
+    - start_block: first block at which snapshot will be taken, if not set, will be taken immediately similar to an old create_snapshot call
+    - end_block: block number at which schedule for the snapshot will be no longer active and snapshot task will be automatically removed. If end_block is not specified, snapshot task will never be auto removed and needs to be unscheduled with another api call to stop
+    - blocks_count - number of blocks after which a recurring snapshot will be made. If not set, snapshot will be done only once and scheduled task will be removed
```

Comment:
`block_count` name is not that clear to users. I got its meaning from reading the github issue. Can it be renamed so it is more clear? At first I thought it was `end_block - start_block + 1`.

Reply:
Renamed